app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/securities/ConfirmFullRepaymentController.scala

```diff
@@ -16,18 +16,142 @@
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.Forms.confirmFullRepaymentForm
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.JourneyControllerComponents
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.mixins.WorkInProgressMixin
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.{routes => baseRoutes}
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.journeys.SecuritiesJourney
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.answers.YesNo
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.answers.YesNo.No
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.answers.YesNo.Yes
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.utils.Logging
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.securities.confirm_full_repayment
 
 import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.Try
 
 @Singleton
 class ConfirmFullRepaymentController @Inject() (
-  val jcc: JourneyControllerComponents
-)(implicit viewConfig: ViewConfig, ec: ExecutionContext)
+  val jcc: JourneyControllerComponents,
+  confirmFullRepaymentPage: confirm_full_repayment
+)(implicit viewConfig: ViewConfig, errorHandler: ErrorHandler, ec: ExecutionContext)
     extends SecuritiesJourneyBaseController
-    with WorkInProgressMixin[SecuritiesJourney]
+    with SecuritiesJourneyRouter
+    with Logging {
+
+  private val form: Form[YesNo] = confirmFullRepaymentForm
+
+  // todo import SecuritiesJourney.Checks._
+
+  // GET          /securities/confirm-full-repayment
+  // @uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.securities.ConfirmFullRepaymentController.showFirst()
+  // @(form: Form[YesNo], securityId: String, totalValue: BigDecimal, postAction: Call)
+  //    (implicit request: RequestWithSessionData[_], messages: Messages, viewConfig: ViewConfig)
+  @SuppressWarnings(Array("org.wartremover.warts.TraversableOps"))
+  def showFirst(): Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    journey.getSecuritiesReclaims.headOption.fold(
+      errorHandler.errorResult().asFuture
+    )(reclaims => showForId(request, journey, reclaims._1))
+  }
+
+  // GET          /securities/confirm-full-repayment/:id
+  // @uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.securities.ConfirmFullRepaymentController.show(id: String)
+  // @(form: Form[YesNo], securityId: String, totalValue: BigDecimal, postAction: Call)
+  //    (implicit request: RequestWithSessionData[_], messages: Messages, viewConfig: ViewConfig)
+  def show(id: String): Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    showForId(request, journey, id)
+  }
+
+  private def showForId(implicit request: Request[_], journey: SecuritiesJourney, id: String) =
+    journey
+      .getDisplayDeclarationIfValidSecurityDepositId(id)
+      .flatMap(_.getSecurityDetailsFor(id).map(_.amountPaid))
+      .flatMap(x => Try(BigDecimal(x)).toOption)
```

Comment:
I would keep it simple here and move BigDecimal into declaration, we will later refactor ACC14 model to have BigDecimals anyway

Reply:
I'll use the new function that was added `getSecurityPaidValueFor`

Reply:
updated to use `getSecurityTotalValueFor` as discussed (currently this method is backed by `totalAmount`)
```scala
 private def showForId(implicit request: Request[_], journey: SecuritiesJourney, id: String) =
    journey
      .getDisplayDeclarationIfValidSecurityDepositId(id)
      .map(_.getSecurityTotalValueFor(id))
      .fold(errorHandler.errorResult()) { amountPaid =>
```